php/coauthors-install.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * Install functions.
+ *
+ * @package Co-Authors Plus
+ */ 
+
+/**
+ * Runs on plugin activation.
+ */ 
+function cap_install_setup() {
+	wp_schedule_single_event( time(), 'cap_import_existing_users' );
+}
+
+
+/**
+ * Creates author terms for existing users.
+ *
+ * Since CAP 2.7, we're searching against the term description for the fields instead of the user details.
+ * When CAP is first installed, terms are missing for all existing users.
+ * This function take care of creating terms for all users who have permission to be added as coauthors.
+ * Uses a WP-Cron event that gets rescheduled until all users have been processed.
+ *
+ * @param int $imported_count number of users already processed (used as `offset` in get_users)
+ * @param int $number_to_update number of users to process per each cron event
+ */ 
+function cap_create_author_terms( $imported_count = 0, $number_to_update = 100 ) {
+	global $coauthors_plus;
+
+	$args = array(
+		'orderby' => 'registered',
+		'order' => 'ASC',
+		'offset' => $imported_count,
+		'number' => $number_to_update,
+		'fields' => 'all_with_meta',
+	);
+	$found_users = get_users( $args );
```

Comment:
I suspect that we might want to test this query. I have a feeling that if this query were ever to be run on a very large site it could break things since there is no index on registered. This means MySQL would have to try to do a table sort on all the xyz number of users. I'm also unsure of how this is handled in a multi-site, clearly the UI can scale but those are ordered by user_nicename which is indexed. I think we should have a look at the types of queries this generates  (and potentially even benchmarking it) before merging.

Reply:
> (Side note: sorting by registration time ASC ... is just an extra-safety measure, because all users created after CAP installation would have their term created as per 2. ).

**tl;dr I can sort by `user_nicename` and nothing bad will happen.**

Just to give you a bit of context on the _why_, I implemented this function _before_ the one that updated author term on profile update. So it seemed neat to make it process in a way that would cover all of them, even if some new ones popped up after the first cron had run. Then I implemented `action_user_profile_update()` and the kind of sorting here became irrelevant. But I decided to keep it because I thought it was neat, except it wasn't because I didn't know about the index thing you mentioned.

_Moral: I fell into the premature-optimization/smarter-but-stupid trap :D_